core/arch/riscv/include/rpmi.h

```diff
@@ -0,0 +1,114 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2025 NXP
+ */
+
+#ifndef __RPMI_H
+#define __RPMI_H
+
+#ifndef __ASSEMBLER__
+
+#include <compiler.h>
+#include <stdint.h>
+
+/* RPMI error codes */
+enum rpmi_error_codes {
+	RPMI_SUCCESS			= 0,
+	RPMI_ERR_FAILED			= -1,
+	RPMI_ERR_NOTSUPP		= -2,
+	RPMI_ERR_INVALID_PARAM		= -3,
+	RPMI_ERR_DENIED			= -4,
+	RPMI_ERR_INVALID_ADDR		= -5,
+	RPMI_ERR_ALREADY		= -6,
+	RPMI_ERR_EXTENSION		= -7,
+	RPMI_ERR_HW_FAULT		= -8,
+	RPMI_ERR_BUSY			= -9,
+	RPMI_ERR_INVALID_STATE		= -10,
+	RPMI_ERR_BAD_RANGE		= -11,
+	RPMI_ERR_TIMEOUT		= -12,
+	RPMI_ERR_IO			= -13,
+	RPMI_ERR_NO_DATA		= -14,
+	RPMI_ERR_RESERVED_START		= -15,
+	RPMI_ERR_RESERVED_END		= -127,
+	RPMI_ERR_VENDOR_START		= -128,
+};
+
+/* RPMI message header */
+struct rpmi_message_header {
+	/* Service group ID */
```

Comment:
Normally we use the pattern at for instance:
https://github.com/OP-TEE/optee_os/blob/614b28146e96edda3e92c316eb78b592e00072c2/core/arch/arm/include/mm/core_mmu_arch.h#L111-L118
to document structs. It's nice if we can stick to one style.